bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/fs/watch/JSScriptFileWatcher.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.automation.jsscripting.internal.fs.watch;
+
+import java.io.File;
+import java.util.Optional;
+
+import org.openhab.automation.jsscripting.internal.GraalJSScriptEngineFactory;
+import org.openhab.core.automation.module.script.ScriptEngineManager;
+import org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileReference;
+import org.openhab.core.automation.module.script.rulesupport.loader.ScriptFileWatcher;
+import org.openhab.core.service.ReadyService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Monitors <openHAB-conf>/automation/js for Javascript files
+ *
+ * @author Jonathan Gilbert - Initial contribution
+ */
+@Component(immediate = true)
+public class JSScriptFileWatcher extends ScriptFileWatcher {
+    private static final String FILE_DIRECTORY = "automation" + File.separator + "js";
```

Comment:
Since this directory does not exist yet, should we attempt to create it when this bundle is activated?

Reply:
So I had a quick look at the code to do this and it raised some questions:

 - This is just an implementation of ScriptFileWatcher, so should this base class create all the directories it watches? This would be `automation/js` and `automation/jsr223`.
 - There is also a ScriptLibraryWatcher, which watches `automation/js/node_modules` - should it create that directory?

I am leaning towards creating the two directories, so putting the creation into ScriptFileWatcher. `node_modules` would get created by npm anyway.

Reply:
Sounds good!

Reply:
I've added it to the PR in core: https://github.com/openhab/openhab-core/pull/2593

Reply:
Awesome !